cli/cli-probe-server/src/main/java/io/telicent/smart/cache/cli/probes/HealthProbeServer.java

```diff
@@ -67,21 +83,60 @@ protected ServerBuilder buildServer() {
      * Runs the server and registers a shutdown hook for cleaning up the server
      */
     public void run() {
-        this.run(false);
+        if (this.future != null) {
+            throw new IllegalStateException("Health Probe Server is already running");
+        }
+        LOGGER.info("Starting Health Probe Server on port {}", this.port);
 
-        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-            if (this.server != null) {
-                this.server.shutdownNow();
+        // Launch this on a background daemon thread
+        this.future = this.executorService.submit(() -> {
+            try {
+                Thread.currentThread().setName("HealthProbeServer");
+            } catch (Throwable e) {
+                // Ignore if unable to set thread name
             }
-        }));
+
+            this.run(true);
+        });
+
+        // Wait briefly to ensure that the server has time to actually start before the caller tries to use it!
+        try {
+            Thread.sleep(250);
```

Comment:
Do we want to configure this? Probably not, but perhaps a final static long value at top of class so that it's obvious? Also means we could potentially reduce it for Testing's sake.

Reply:
Yeah so this was added due to testing issues, what might be better would be to do the wait in the tests because for general users they don't care if the server is available immediately as long as it can service health probes eventually

Reply:
Moved this out to the tests